### PR TITLE
Useful hedgehog annotations on CLI command failure

### PR DESCRIPTION
**Pull request by newhoggy**
This improves the `hedgehog` output when invocation of the CLI fails.

Previously CLI failure looked like this:

```
  ● golden_shelleyStakeAddressRegistrationCertificate passed 0 tests (running)
  ○ 16/25 complete (running)
Invalid option `--out-files'

Did you mean this?
    --out-file

Usage: cardano-cli shelley stake-address registration-certificate --stake-verification-key-file FILE
  ✗ golden_shelleyStakeAddressRegistrationCertificate failed at test/Test/OptParse.hs:244:33
    after 1 test.
  
        ┏━━ test/Test/OptParse.hs ━━━
    243 ┃ propertyOnce :: H.PropertyT IO () -> H.Property
    244 ┃ propertyOnce =  H.withTests 1 . H.property
        ┃ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        ┃ │ ━━━ Exception (IOException) ━━━
        ┃ │ readCreateProcess: cabal "exec" "--" "cardano-cli" "shelley" "stake-address" "registration-certificate" "--staking-verification-key-file" "test/Test/golden/shelley/keys/stake_keys/verification_key" "--out-files" "tmp/stake-address-registration-certificate/test-08dd7abc55943679/registration.cert" (exit 1): failed
```

Above the `stderr` is arbitrarily interleaved in with other test output, and the error is not particularly useful and not associated with the failing test case.

Now the CLI failure looks like this, showing clearly, the command, stdout, stderr, and exit code:

```
  ✗ golden_shelleyStakeAddressRegistrationCertificate failed at test/Test/CLI/Shelley/Golden/StakeAddress/RegistrationCertificate.hs:21:12
    after 1 test.
  
       ┏━━ test/Test/CLI/Shelley/Golden/StakeAddress/RegistrationCertificate.hs ━━━
    15 ┃ golden_shelleyStakeAddressRegistrationCertificate :: Property
    16 ┃ golden_shelleyStakeAddressRegistrationCertificate = OP.propertyOnce $ do
    17 ┃   OP.workspace "tmp/stake-address-registration-certificate" $ \tempDir -> do
       ┃   │ Workspace: cardano-cli/tmp/stake-address-registration-certificate/test-480be8dbfbc65e5c
    18 ┃     keyGenStakingVerificationKeyFile <- OP.noteInputFile "test/Test/golden/shelley/keys/stake_keys/verification_key"
       ┃     │ cardano-cli/test/Test/golden/shelley/keys/stake_keys/verification_key
    19 ┃     registrationCertFile <- OP.noteTempFile tempDir "regi stration.cert"
       ┃     │ cardano-cli/tmp/stake-address-registration-certificate/test-480be8dbfbc65e5c/regi stration.cert
    20 ┃ 
    21 ┃     void $ OP.execCardanoCLI
    22 ┃         [ "shelley","stake-address","registration-certificate"
    23 ┃         , "--staking-verification-key-file", keyGenStakingVerificationKeyFile
    24 ┃         , "--out-files", registrationCertFile
    25 ┃         ]
       ┃         ^
       ┃         │ Process exited with non-zero exit-code
       ┃         │ ━━━━ command ━━━━
       ┃         │ cardano-cli shelley stake-address registration-certificate --staking-verification-key-file test/Test/golden/shelley/keys/stake_keys/verification_key --out-files "tmp/stake-address-registration-certificate/test-3a457631a893b18b/regi stration.cert"
       ┃         │ ━━━━ stdout ━━━━
       ┃         │ 
       ┃         │ ━━━━ stderr ━━━━
       ┃         │ Invalid option `--out-files'
       ┃         │ 
       ┃         │ Did you mean this?
       ┃         │     --out-file
       ┃         │ 
       ┃         │ Usage: cardano-cli shelley stake-address registration-certificate --stake-verification-key-file FILE
       ┃         │                                                                   --out-file FILE
       ┃         │   Create a stake address registration certificate
       ┃         │ 
       ┃         │ ━━━━ exit code ━━━━
       ┃         │ 1
    26 ┃ 
    27 ┃     OP.assertFileOccurences 1 "Stake Address Registration Certificate" registrationCertFile
  
    This failure can be reproduced by running:
    > recheck (Size 0) (Seed 6318384335668956750 240684346131424873) golden_shelleyStakeAddressRegistrationCertificate
```

### CODE REVIEW

File: cardano-cli/test/Test/golden/shelley/node-pool/kes.skey [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (intricate) [inline cardano-cli/test/Test/golden/shelley/node-pool/kes.skey:1]: What are all of these new golden files for? Are they related to this PR?

Comment (newhoggy) [reply cardano-cli/test/Test/golden/shelley/node-pool/kes.skey:1]: Oops.  They weren't supposed to be included.  Will remove.

File: cardano-cli/test/Test/OptParse.hs [hunk 2]
```diff
@@ -58,19 +61,52 @@ import           Hedgehog.Internal.Source (getCaller)
 evalCardanoCLIParser :: [String] -> Opt.ParserResult ClientCommand
 evalCardanoCLIParser args = Opt.execParserPure pref opts args
 
+-- | Format argument for a shell CLI command.
+--
+-- This includes automatically embedding string in double quotes if necessary, including any necessary escaping.
+--
+-- Note, this function does not cover all the edge cases for shell processing, so avoid use in production code.
```
Comment (intricate) [inline cardano-cli/test/Test/OptParse.hs:68]: :100:

**Comment by newhoggy** [comment]
bors merge
